hexgame_ai/tree_creator.py

```diff
@@ -0,0 +1,28 @@
+from hexgame.color import Color
+from hexgame.board import Board
+import copy
+
+
+class NodeTree:
+    def expand_tree_rec(board: Board,
+                        action: tuple[int, int], depth: int):
+        # print("-------------------recursion-------------------\n")
+        # print(f"board:\n {board}"
+        #       f"action: {action}\n"
+        #       f"depth: {depth}\n")
+        if action:
+            current_color = Color.Red if depth % 2 == 1 else Color.Blue
+            board.place_stone(
+                action[0], action[1], current_color
+            )
+            if board._has_color_won(current_color):
+                # print(f"player {current_color} has won\n")
+                return {"data": action, "children": []}
+        possible_actions = board.empty_positions
+        # print(f"possible actions:{possible_actions}\n")
+        children = []
+        for a in possible_actions:
+            next_board = copy.deepcopy(board)
```

Comment:
I just worry about this whole implementation how practical you think it is to use in real scenarios. I.e., surely this will never be able to keep an actual whole tree right? I.e., do you think it is actually feasible to use this into the MCTS we plan to define?

Reply:
We will try to face those issues once they come up, I guess.